provider/github_test.go

```diff
@@ -8,28 +8,48 @@ import (
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/assert"
 	"io/ioutil"
+	"sync"
 	"testing"
 )
 
 func initLogger() {
 	logrus.SetLevel(7)
 }
 
+const defaultTag = "test-tag"
+
 type MockPullRequestService struct {
+	sync.RWMutex
 	comments []*github.IssueComment
 }
 
 func (m *MockPullRequestService) DeleteComment(ctx context.Context, owner string, repo string, commentID int64) (*github.Response, error) {
+	var index int
+	var found bool
+	for i, comment := range m.comments {
+		if *comment.ID == commentID {
+			index = i
+			found = true
+		}
+	}
+	if !found {
+		return nil, fmt.Errorf("could not find comment to delete with id %d", commentID)
+	}
+	copy(m.comments[index:], m.comments[index+1:])
```

Comment:
hard to read and understand what L38-40 is doing. (At least I always forget this slice magic). Maybe wrap it in an inline function giving the variable a meaningful name of what is done
```
theMagic := func(...) {
    L38-L40
}

theMagic(...)
```

Reply:
yes I can source that out to a method, good point, it's code to remove an element from a slice, interesting that golang does not have an inbuilt function for that

Reply:
testing function no urgent need to fix, but I will keep when used multiple times